test/common/router/config_impl_test.cc

```diff
@@ -2128,6 +2132,40 @@ TEST(RouteMatcherTest, TestDuplicateDomainConfig) {
                EnvoyException);
 }
 
+TEST(RouteMatcherTest, TestCaseSensitiveDomainConfig) {
+  std::string json = R"EOF(
+{
+  "virtual_hosts": [
+    {
+      "name": "www2",
+      "domains": ["www.lyft.com"],
+      "routes": [
+        {
+          "prefix": "/",
+          "cluster": "www2"
+        }
+      ]
+    },
+    {
+      "name": "www2_staging",
+      "domains": ["www.LYFt.com"],
+      "routes": [
+        {
+          "prefix": "/",
+          "cluster": "www2_staging"
+        }
+      ]
+    }
+  ]
+}
+  )EOF";
+
+  NiceMock<Runtime::MockLoader> runtime;
+  NiceMock<Upstream::MockClusterManager> cm;
+  EXPECT_THROW(ConfigImpl config(parseRouteConfigurationFromJson(json), runtime, cm, true),
```

Comment:
As this is a new test, can you use EXPECT_THROW_WITH_MESSAGE to verify the actual error message?

Reply:
Makes sense. Will do it.